clean/hammond_pd_cprr.py

```diff
@@ -0,0 +1,149 @@
+        .str.replace('7/2010 til 5/2012', '07/01/2010', regex=False)
+    return df
+
+
+def clean_disposition(df):
+    df.loc[:, 'disposition'] = df.finding.fillna('').str.lower().str.strip()\
+        .str.replace('/', ' | ')\
+        .str.replace('no action', '', regex=False)
+    return df.drop(columns='finding')
+
+
+def split_name(df):
+    names = df.ee_name.str.lower().str.strip().str.extract(r'^(\w+) (\w+)')
+    df.loc[:, 'first_name'] = names[0]
+    df.loc[:, 'last_name'] = names[1]
+    return df.drop(columns='ee_name')
+
+
+def clean_department_desc(df):
+    df.loc[:, 'department_desc'] = df.division.str.lower().str.strip()\
+        .str.replace('jailer', 'corrections', regex=False)
+    return df.drop(columns=['dept', 'division'])
+
+
+def clean_action(df):
+    df.loc[:, 'action'] = df.final_recommendation.fillna('').str.lower().str.strip()\
+        .str.replace(
+            'three (3) consecutive shift-days suspenion without pay',
+            '3-day suspension without pay', regex=False)\
+        .str.replace(
+            '5-five consecutive shift-days suspension without pay',
+            '5-day suspension without pay', regex=False)\
+        .str.replace(
+            '| day suspension commence 1/28/17',
+            'suspended', regex=False)\
+        .str.replace('days suspension 2 days', '2-day suspension', regex=False)\
+        .str.replace(r'^(\w+) resigned', 'resigned', regex=True)\
+        .str.replace('suspended without pay period of 3 days', '3-day suspension', regex=False)\
+        .str.replace('frrom', 'from', regex=False)\
+        .str.replace('edwin bergeron 10/2/2019', '', regex=False)\
+        .str.replace(r'^(\d+) (\w+)', r'\1-\2', regex=True)\
+        .str.replace(' / ', ' | ', regex=False)
+    return df.drop(columns='final_recommendation')
+
+
+def combine_columns(df):
+    def combine(row):
+        txts = []
+        if pd.notnull(row.violation_1):
+            txts.append('%s' % row.violation_1)
+        if pd.notnull(row.violation_2_3):
+            txts.append('%s' % row.violation_2_3)
+        return '| '.join(txts)
+    df.loc[:, 'charges'] = df.apply(combine, axis=1, result_type='reduce')
+    df = df.drop(columns=['violation_1', 'violation_2_3'])
+    return df
+
+
+def clean_charges(df):
+    df.charges = df.charges.str.lower().str.strip().fillna('')\
+        .str.replace(r'^(\d+) (\w+)', r'\1 - \2', regex=True)\
+        .str.replace('.', '', regex=False)\
+        .str.replace(r' \bgo\b ', ' ', regex=True)\
+        .str.replace('/', ' and ', regex=False)\
+        .str.replace(r'\(founded\)', '', regex=True)\
+        .str.replace(r'^excessive force ', 'excessive force | ', regex=True)\
+        .str.replace('omissionn', 'omission', regex=False)\
+        .str.replace(r' ?(unbec?o?i?mo?ing) ?( ?(of)? ?(an officer)?)? ?', ' unbecoming ', regex=True)\
+        .str.replace('video conduct unbecoming', 'video | conduct unbecoming', regex=False)
+    return df

```

Comment:
Nothing wrong with standardizing using `str.replace` but for tasks like this try using `standardize_from_lookup_table` sometime to get a feel of how it works. I feel that this is a powerful tool that has been forgotten about for too long.

Reply:
I'm using the standardize_look_up for Baton Rouge PD now. I wanted to get a bit more practice with regex w/ hammond_pd